datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -1128,10 +1137,37 @@ fn finalize_aggregation(
         | AggregateMode::Single
         | AggregateMode::SinglePartitioned => {
             // merge the state to the final value
-            accumulators
+            let res = accumulators
                 .iter()
-                .map(|accumulator| accumulator.evaluate().map(|v| v.to_array()))
-                .collect::<Result<Vec<ArrayRef>>>()
+                .map(|accumulator| 
+                {
+                    println!("mode: {:?}", mode);
+                    let res = accumulator.evaluate_v2();
```

Comment:
I wonder if you could do

```suggestion
                    let res = accumulator.evaluate()?;
                    res.to_array()
                    
```

Reply:
The problem is that I need to find a way to represent Array (Column support) with ScalarValue. So that I can get the ArrayRef from ScalarValue `to_array`. ScalarValue::List does not fit at all.